presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestRFC2822JsonDateTimeFormatter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json;
+
+import io.prestosql.plugin.kafka.encoder.json.format.DateTimeFormat;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+import io.prestosql.spi.type.TimeZoneKey;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.prestosql.spi.type.TimeZoneKey.UTC_KEY;
+import static io.prestosql.testing.DateTimeTestingUtils.sqlTimestampOf;
+import static io.prestosql.testing.DateTimeTestingUtils.sqlTimestampWithTimeZoneOf;
+import static io.prestosql.testing.assertions.Assert.assertEquals;
+
+public class TestRFC2822JsonDateTimeFormatter
+{
+    private static final DateTimeFormat DATE_TIME_FORMAT = DateTimeFormat.RFC2822;
+
+    private void testTimestamp(SqlTimestamp value, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.empty()).formatTimestamp(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    private void testTimestampWithTZ(SqlTimestampWithTimeZone value, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.empty()).formatTimestampWithZone(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    @Test
+    public void testRFC2822DateTimeFunctions()
+    {
+        testTimestamp(sqlTimestampOf(3, 2020, 8, 18, 12, 38, 29, 0), "Tue Aug 18 12:38:29 +0000 2020");
+        testTimestamp(sqlTimestampOf(3, 1970, 1, 1, 0, 0, 0, 0), "Thu Jan 01 00:00:00 +0000 1970");
+        testTimestamp(sqlTimestampOf(3, 1800, 8, 18, 12, 38, 29, 0), "Mon Aug 18 12:38:29 +0000 1800");
+
+        testTimestampWithTZ(sqlTimestampWithTimeZoneOf(3, 2020, 8, 19, 12, 23, 41, 0, UTC_KEY), "Wed Aug 19 12:23:41 +0000 2020");
+        testTimestampWithTZ(sqlTimestampWithTimeZoneOf(3, 2020, 8, 19, 12, 23, 41, 0, TimeZoneKey.getTimeZoneKey("America/New_York")), "Wed Aug 19 12:23:41 +0000 2020");
```

Comment:
same question about timezone. Why are we always going for UTC and do not use the TZ from value?